forem/forem
# Point changes in notification settings to `users_notification_settings` table

## Description
This PR points changes to notification settings to the `users_notification_settings` table. Changes previously pointed to the `users` table.
This PR shall be merged into [PR 13904](https://github.com/forem/forem/pull/13904).

## TO-DO list for this PR
- [x] Update the `settings/notifications` tab to use the new `users_notification_settings` table
- [x] Delete the notification_settings-related sync code in `user.rb` 
- [x] Check for other instances where these fields are being used


## What type of PR is this? (check all applicable)





## QA Instructions, Screenshots, Recordings
1. Sign into `localhost`; in rails console, pull up the same user that you are signed in as.
2. In rails console, run `user.notification_setting` and take note the value for `email_badge_notifications`. Also run `user.email_badge_notifications` and note that its value matches that in the `notification_setting` record. 
3. In the app, navigate to `localhost:xxxx/settings/notifications` and check or uncheck `Send me an email when I receive a badge`. Click Save. You should be redirected to the Notifications Tab, and the box you checked should still be checked.
4.  In the console, reload your user (`user.reload`) and re-run both` user.notification_setting` and `user.email_badge_notifications`. Now, the 2 values should differ, as the `users` record was left unchanged, but the `notification_settings` record was changed accordingly. 


### UI accessibility concerns? NONE (backend changes)

## Added tests?

      have not been included_
- [X] Not yet; will write/fix tests after merge into base PR 13904

## [Forem core team only] How will this change be communicated?
